src/components/TracePage/ArchiveNotifier/duck.js

```diff
@@ -0,0 +1,83 @@
+// @flow
```

Comment:
why is it called `duck.js`?

Reply:
Seems like the name of this approach (combining the reducer and actions into a single file) is a play on word(s) involving redux: https://github.com/erikras/ducks-modular-redux 

Derivative stuff uses the ducks word: https://github.com/erikras/ducks-modular-redux#implementation